Gems/Atom/RHI/Code/Source/RHI/BufferPool.cpp

```diff
@@ -155,20 +155,80 @@ namespace AZ::RHI
             initRequest.m_descriptor,
             [this, &initRequest]()
             {
-                return IterateObjects<DeviceBufferPool>([&initRequest](auto deviceIndex, auto deviceBufferPool)
+                initRequest.m_buffer->Init(GetDeviceMask() & initRequest.m_deviceMask);
+
+                return IterateObjects<DeviceBufferPool>(
+                    [&initRequest](auto deviceIndex, auto deviceBufferPool)
+                    {
+                        bool initBuffer = ((AZStd::to_underlying(initRequest.m_buffer->GetDeviceMask()) >> deviceIndex) & 1) == 1;
+
+                        if (!initRequest.m_buffer->m_deviceObjects.contains(deviceIndex))
+                        {
+                            if (initBuffer)
+                            {
+                                initRequest.m_buffer->m_deviceObjects[deviceIndex] = Factory::Get().CreateBuffer();
+                            }
+                        }
+                        else
+                        {
+                            if (!initBuffer)
+                            {
+                                initRequest.m_buffer->m_deviceObjects.erase(deviceIndex);
+                            }
+                        }
+
+                        if (initBuffer)
+                        {
+                            DeviceBufferInitRequest bufferInitRequest(
+                                *initRequest.m_buffer->GetDeviceBuffer(deviceIndex), initRequest.m_descriptor, initRequest.m_initialData);
+                            return deviceBufferPool->InitBuffer(bufferInitRequest);
+                        }
+
+                        return ResultCode::Success;
+                    });
+            });
+
+        return resultCode;
+    }
+
+    ResultCode BufferPool::UpdateBufferDeviceMask(const BufferDeviceMaskRequest& request)
+    {
+        AZ_PROFILE_FUNCTION(RHI);
```

Comment:
Should there be a check (and possible warning) to see if the `BufferPool` even supports this deviceMask? As far as I can see, trying to set a device which is not present in the pools will not elicit any warning (and then crash later probably).

Reply:
The loop only goes over the pool's devices, so only those bits are considered. Where exactly would it crash?